packages/expo-gl-cpp/cpp/EXGLNativeMethods.cpp

```diff
@@ -0,0 +1,1558 @@
+      if (requiresWebGL2 && !exglCtx->supportsWebGL2) {                                                 \
+        throw std::runtime_error("EXGL: This device doesn't support WebGL2 method: " #name "()!");      \
+      }                                                                                                 \
+      return exglCtx->exglNativeInstance_##name(jsCtx, jsFunction, jsThis,                              \
+                                                jsArgc, jsArgv, jsException);                           \
+    } catch (const std::exception &e) {                                                                 \
+      exglCtx->jsThrow(jsCtx, e.what(), jsException);                                                   \
+      return nullptr;                                                                                   \
+    }                                                                                                   \
+  }                                                                                                     \
+  JSValueRef EXGLContext::exglNativeInstance_##name(JSContextRef jsCtx,                                 \
+                                              JSObjectRef jsFunction,                                   \
+                                              JSObjectRef jsThis,                                       \
+                                              size_t jsArgc,                                            \
+                                              const JSValueRef jsArgv[],                                \
+                                              JSValueRef* jsException)
+
+#define _WRAP_METHOD(name, minArgc) _WRAP_METHOD_INTERNAL(name, minArgc, false)
+#define _WRAP_WEBGL2_METHOD(name, minArgc) _WRAP_METHOD_INTERNAL(name, minArgc, true)
+
+  // Wrapper raises an exception saying the function isn't implemented yet
+#define _WRAP_METHOD_UNIMPL(name)                                       \
+  _WRAP_METHOD(name, 0) {                                               \
+    throw std::runtime_error("EXGL: " #name "() isn't implemented yet!"); \
+    return nullptr;                                                     \
+  }
+
+  // Wrapper that takes only scalar arguments and returns nothing
+#define _WRAP_METHOD_SIMPLE_INTERNAL(name, isWebGL2Method, glFunc, ...) \
+  _WRAP_METHOD_INTERNAL(name, EXJS_ARGC(__VA_ARGS__), isWebGL2Method) { \
+    addToNextBatch(std::bind(glFunc, EXJS_MAP_EXT(0, _EXJS_COMMA, _WRAP_METHOD_SIMPLE_UNPACK, __VA_ARGS__))); \
+    return nullptr;                                                     \
+  }
+#define _WRAP_METHOD_SIMPLE(name, glFunc, ...) _WRAP_METHOD_SIMPLE_INTERNAL(name, false, glFunc, __VA_ARGS__)
+#define _WRAP_WEBGL2_METHOD_SIMPLE(name, glFunc, ...) _WRAP_METHOD_SIMPLE_INTERNAL(name, true, glFunc, __VA_ARGS__)
+
+#define _WRAP_METHOD_SIMPLE_UNPACK(i, _) EXJSValueToNumberFast(jsCtx, jsArgv[i])
+
+
+  // This listing follows the order in
```

Comment:
Is there any specific reason for indenting almost whole file with 2 spaces?

Reply:
it was copy paste from inside class

Reply:
👍 to @bbarthec, since it is a new file we could just make the indentation as it should be